### ISSUE TITLE
Underflow bug in :chat

### ISSUE THREAD

**Issue by custom-jonathan** [urbit/urbit#1864]
New favorite error from my client on my backup ship:

```
peek slam fail
subtract-underflow
/~talleg-botsug/home/0/app/chat-store/hoon:<[103 29].[103 44]>
/~talleg-botsug/home/0/app/chat-store/hoon:<[103 22].[103 45]>
/~talleg-botsug/home/0/app/chat-store/hoon:<[103 16].[103 56]>
/~talleg-botsug/home/0/app/chat-store/hoon:<[103 5].[103 57]>
/~talleg-botsug/home/0/app/chat-store/hoon:<[102 5].[103 57]>
/~talleg-botsug/home/0/app/chat-store/hoon:<[100 5].[103 57]>
/~talleg-botsug/home/0/app/chat-store/hoon:<[99 5].[103 57]>
/~talleg-botsug/home/0/app/chat-store/hoon:<[94 5].[103 57]>
/~talleg-botsug/home/0/app/chat-store/hoon:<[93 5].[103 57]>
/~talleg-botsug/home/0/app/chat-store/hoon:<[92 5].[103 57]>
/~talleg-botsug/home/0/app/chat-store/hoon:<[78 5].[103 57]>
/~talleg-botsug/home/0/app/chat-store/hoon:<[77 5].[103 57]>
/~talleg-botsug/home/0/app/chat-store/hoon:<[75 5].[103 57]>
/~talleg-botsug/home/0/app/chat-store/hoon:<[74 5].[103 57]>
/~talleg-botsug/home/0/app/chat-store/hoon:<[73 5].[103 57]>
/~talleg-botsug/home/0/app/chat-store/hoon:<[71 3].[104 5]>
/~talleg-botsug/home/0/app/chat-store/hoon:<[70 3].[104 5]>
~talleg-botsug:chat-cli> 
```

**Comment by philipcmonk** [comment]
@loganallenc

**Comment by belisarius222** [comment]
I just saw this error on ~rovnys.  I haven't noticed any symptoms other than the stack trace.

**Comment by belisarius222** [comment]
https://github.com/urbit/urbit/blob/master/pkg/arvo/app/chat-store.hoon#L103

Maybe an arithmetic bug?

**Comment by custom-jonathan** [comment]
Whatever it is, I just saw it again for the first time since then that I've noticed:

```
peek slam fail
subtract-underflow
/~minder-folden/home/0/app/chat-store/hoon:<[103 29].[103 44]>
/~minder-folden/home/0/app/chat-store/hoon:<[103 22].[103 45]>
/~minder-folden/home/0/app/chat-store/hoon:<[103 16].[103 56]>
/~minder-folden/home/0/app/chat-store/hoon:<[103 5].[103 57]>
/~minder-folden/home/0/app/chat-store/hoon:<[102 5].[103 57]>
/~minder-folden/home/0/app/chat-store/hoon:<[100 5].[103 57]>
/~minder-folden/home/0/app/chat-store/hoon:<[99 5].[103 57]>
/~minder-folden/home/0/app/chat-store/hoon:<[94 5].[103 57]>
/~minder-folden/home/0/app/chat-store/hoon:<[93 5].[103 57]>
/~minder-folden/home/0/app/chat-store/hoon:<[92 5].[103 57]>
/~minder-folden/home/0/app/chat-store/hoon:<[78 5].[103 57]>
/~minder-folden/home/0/app/chat-store/hoon:<[77 5].[103 57]>
/~minder-folden/home/0/app/chat-store/hoon:<[75 5].[103 57]>
/~minder-folden/home/0/app/chat-store/hoon:<[74 5].[103 57]>
/~minder-folden/home/0/app/chat-store/hoon:<[73 5].[103 57]>
/~minder-folden/home/0/app/chat-store/hoon:<[71 3].[104 5]>
/~minder-folden/home/0/app/chat-store/hoon:<[70 3].[104 5]>
```

**Comment by joemfb** [comment]
@loganallenc, the +peek arm here is rolling its own signed arithmetic, but it should be able to use `@sd` and the operations from `+si`.

**Comment by tacryt-socryp** [comment]
Fixed in #4058.
